### PR TITLE
Issue #3324

### PR DESCRIPTION
**Pull request by bitfurry**
The issue was that _id2ref function was retruing nil value. I fixed it to throw a RangeError. Please correct me if I am wrong at any point. 

jruby_pullrequest

### CODE REVIEW

**Comment by kares** [comment]
najs, would you mind cleaning-up the commits ? `git squash` them into one and maybe avoid touching *truffle/src/main/ruby/core/rubinius/common/object_space.rb* as the truffle part is unaffected ...

File: core/src/main/java/org/jruby/RubyObjectSpace.java [hunk 1]
```diff
@@ -114,7 +114,7 @@ public static IRubyObject id2ref(IRubyObject recv, IRubyObject id) {
                 return object;
             } else {
                 runtime.getWarnings().warn("ObjectSpace is disabled; _id2ref only supports immediates, pass -X+O to enable");
-                return runtime.getNil();
+                throw recv.getRuntime().newRangeError(id+" This is not id value");
```
Comment (chrisseaton) [inline core/src/main/java/org/jruby/RubyObjectSpace.java]: Can you try to replicate the error message exactly? It should be `0x00000000000000 is not id value`, which you can probably make with `String.format('%016x is not id value', id)`.

File: truffle/src/main/ruby/core/rubinius/common/object_space.rb [hunk 2]
```diff
@@ -35,8 +35,7 @@ def self._id2ref(id)
     if find_object([:object_id, Integer(id)], ary) > 0
       return ary.first
     end
-
-    return nil
+  return nil
```
Comment (chrisseaton) [inline truffle/src/main/ruby/core/rubinius/common/object_space.rb]: This is a valid commit, but it should be separate from this PR.

File: core/src/main/java/org/jruby/RubyObjectSpace.java (hunk 1 above)
Comment (bitfurry) [inline core/src/main/java/org/jruby/RubyObjectSpace.java]: @chrisseaton I will sure.

File: truffle/src/main/ruby/core/rubinius/common/object_space.rb (hunk 2 above)
Comment (bitfurry) [inline truffle/src/main/ruby/core/rubinius/common/object_space.rb]: Okay, I will create a seperate PR.

**Comment by nirvdrum** [comment]
@bitfurry Thanks for the patch! In the future, please include the issue title or some other context in the commit message. We have a fair number of people that use the JRuby repository without constant access to GitHub. While "issue #3324" links great on GitHub site, it's far less usable from the command-line or non-GitHub mirrors.

**Comment by chrisseaton** [comment]
The failure looks unrelated, and I'll merge now rather than pestering you about the unrelated line again.

Thanks very much for your contribution!

File: truffle/src/main/ruby/core/rubinius/common/object_space.rb (hunk 2 above)
Comment (eregon) [inline truffle/src/main/ruby/core/rubinius/common/object_space.rb]: How is this valid?

Comment (chrisseaton) [inline truffle/src/main/ruby/core/rubinius/common/object_space.rb]: Hmm now you've said that yeah it doesn't make sense. Don't worry - I'll revert it.

Comment (bitfurry) [inline truffle/src/main/ruby/core/rubinius/common/object_space.rb]: @chrisseaton do I need to revert my whole commit or you will revert this part only ?

Comment (chrisseaton) [inline truffle/src/main/ruby/core/rubinius/common/object_space.rb]: Don't worry - I fixed it in 6b911f3a3af6c01adecee5a8ebc72eac81899deb.

Comment (bitfurry) [inline truffle/src/main/ruby/core/rubinius/common/object_space.rb]: Thanks :+1:

**Pull request merged**